File: core/src/net/bossdragon/system/base/AttachmentSystem.java
package net.bossdragon.system.base;

import com.artemis.Aspect;
import com.artemis.Entity;
import com.artemis.systems.EntityProcessingSystem;
import com.badlogic.gdx.math.Vector2;
import net.bossdragon.component.base.Attached;
import net.bossdragon.component.base.Position;
import net.mostlyoriginal.api.plugin.extendedcomponentmapper.M;

/**
 *
 */
public class AttachmentSystem extends EntityProcessingSystem {
    public AttachmentSystem() {
        super(Aspect.all(Attached.class, Position.class));
    }

    M<Attached> mAttached;
    M<Position> mPosition;

    private final Vector2 tmpPos = new Vector2();

    @Override
    protected void process(Entity e) {
        final Attached att = mAttached.get(e);
        final Position pos = mPosition.get(e);
        final Position attPos = mPosition.get(att.entityId);

        tmpPos.set(attPos.desiredPos).add(att.displace);
        pos.xy(tmpPos);
    }
}

File: core/src/net/bossdragon/system/base/AttachmentSystem.kt
package net.bossdragon.system.base

import com.artemis.Aspect
import com.artemis.Entity
import com.artemis.systems.EntityProcessingSystem
import com.badlogic.gdx.math.Vector2
import net.mostlyoriginal.api.plugin.extendedcomponentmapper.M

/**
 *
 */
class AttachmentSystem : EntityProcessingSystem(
    Aspect.all(Attached::class.java, Position::class.java)
) {
    lateinit var mAttached: M<Attached>
    lateinit var mPosition: M<Position>

    val tmpPos = Vector2()

    override fun process(e: Entity) {
        val att = mAttached[e]
        val pos = mPosition[e]
        val attPos = mPosition[att.entityId]

        tmpPos.set(attPos.desiredPos).add(att.displace)
        pos.xy(tmpPos)
    }

}
